app/models/location.rb

```diff
@@ -0,0 +1,118 @@
+class Location
+  include ActiveModel::Validations
+
+  attr_reader :country_code, :subdivision_code
+
+  # TODO: Custom error messages maybe?
+  validates :country_code, inclusion: { in: ISO3166::Country.codes }
+  # TODO: Stronger validation to only allow nil if the country has no subdivisions?
+  validates :subdivision_code, inclusion: {
+    in: ->(location) { ISO3166::Country.new(location.country_code).subdivision_codes }
+  }, allow_nil: true
+
+  def self.from_client_geo(client_geo)
+    return if client_geo.nil?
+
+    country, subdivision = client_geo.split("-")
+
+    new(country, subdivision)
+  end
+
+  # Method to quickly spin up a bunch of records to test against
+  def self.create_test_records(org_id, area: "sidebar_left", count: 500)
```

Comment:
Minor: I imagine this was _just_ for the experiment, but just in case: I think we'd rather test code stayed out of `app/models` ?

Reply:
Yep, this was from me trying to quickly spin up/tear down billboards to poke at while developing. I left it in so folk could run benchmarks themselves, but it's definitely not making it into a proper PR 😅